python/tests/test_rpy_diffusive_flux.py

```diff
@@ -0,0 +1,164 @@
+import numpy as np
+import pytest
+
+import flyft
+from pytest_lazyfixture import lazy_fixture
+
+
+@pytest.fixture
+def spherical_mesh_grand():
+    return flyft.state.SphericalMesh(10.,100, "repeat") 
+
+@pytest.fixture
+def state_grand(spherical_mesh_grand):
+    return flyft.State(flyft.state.ParallelMesh(spherical_mesh_grand), ("A", ))
+
+@pytest.fixture
+def binary_state_grand(spherical_mesh_grand):
+    return flyft.State(flyft.state.ParallelMesh(spherical_mesh_grand), ("A","B"))
+
+def test_ideal(grand,ig,rpy,state_grand):
+    state = state_grand
+
+    volume = state.mesh.full.volume()
+    ig.volumes['A'] = 1.0
+    grand.ideal = ig
+    rpy.diameters['A'] = 1.0
+    rpy.viscosity = 1.0
+
+    # first check OK with all zeros
+    state.fields['A'][:] = 0.
+    grand.constrain('A', 0, grand.Constraint.N)
+    rpy.compute(grand,state)
+    assert np.allclose(rpy.fluxes['A'], 0.)
+
+    # same thing with all ones
+    x = state.mesh.local.centers
+    state.fields['A'][:] = 1
+    grand.constrain('A', volume, grand.Constraint.N)
+    rpy.compute(grand,state)
+    assert np.allclose(rpy.fluxes['A'], 0.)
+    
+    x = state.mesh.local.centers
+    state.fields['A'][:] = 3.0/state.mesh.full.L*(x[:])
+    grand.constrain('A', state.mesh.full.L*1.5, grand.Constraint.N)
+    rpy.compute(grand,state)
+    flux = np.zeros(len(x))
+    for i in range(0,len(x)):
```

Comment:
This is sufficient, the default is to start from 0:
```suggestion
    for i in range(len(x)):
```